### PR TITLE
Updated the expected messages for post experiments

### PR DESCRIPTION
**Pull request by chandrams**
Updated the expected messages for post experiments

### CODE REVIEW

**Comment by dinogun** [comment]
In general I would suggest keeping the test changes and the functionality changes in two different PRs. That will help us track the changes better and also easier to pinpoint issues.

File: src/json_validate.py [hunk 1]
```diff
@@ -123,16 +131,25 @@ def validate_search_space(search_space):
...
+
         # Check if the direction is supported
-        if str(key) == "direction" and str(search_space[key]) not in HPOSupportedTypes.DIRECTIONS_SUPPORTED:
+        elif str(key) == "direction" and str(search_space[key]) not in HPOSupportedTypes.DIRECTIONS_SUPPORTED:
             validationErrorMsg = ",".join([validationErrorMsg, HPOErrorConstants.DIRECTION_NOT_SUPPORTED])
 
         # Check if hpo_algo_impl is supported
-        if str(key) == "hpo_algo_impl" and str(search_space[key]) not in HPOSupportedTypes.ALGOS_SUPPORTED:
+        elif str(key) == "hpo_algo_impl" and str(search_space[key]) not in HPOSupportedTypes.ALGOS_SUPPORTED:
+            print("validationErrorMsg: ", validationErrorMsg)
```
[the remark this answers is not in the document]
Comment (khansaad) [reply src/json_validate.py]: Removed it. Error message is already getting logged outside this function.

**Comment by chandrams** [comment]
Tested the fix, above mentioned issues are fixed now.

File: src/hpo_service.py [hunk 2]
```diff
@@ -33,7 +36,7 @@ def __init__(self):
     def newExperiment(self, id_, experiment_name, total_trials, parallel_trials, direction, hpo_algo_impl,
                       objective_function, tunables, value_type):
         if self.containsExperiment(experiment_name):
-            print("Experiment already exists")
+            logger.error(HPOErrorConstants.EXPERIMENT_EXISTS)
```
Comment (dinogun) [inline src/hpo_service.py:39]: It looks like we are not returning a error value here in `newExperiment` and the calling function goes ahead even if this fails. In addition to the logging the error message we need to return a error value which needs to be checked from the calling function.

File: src/hpo_service.py [hunk 3]
```diff
@@ -69,7 +72,8 @@ def startExperiment(self, name):
             started.release()
 
         if not value:
-            print("Starting experiment timed  out!")
+            logger.error(HPOErrorConstants.EXPERIMENT_TIMED_OUT)
+            return HPOErrorConstants.EXPERIMENT_TIMED_OUT
```
Comment (dinogun) [inline src/hpo_service.py:76]: Can we add a comment here on how the experiment can time out ?

File: src/rest_service.py [hunk 4]
```diff
@@ -247,7 +262,9 @@ def get_search_create_study(search_space_json, operation):
             hpo_service.instance.newExperiment(id_, experiment_name, total_trials, parallel_trials, direction,
                                                hpo_algo_impl, objective_function, tunables, value_type)
             logger.info("Starting Experiment: " + experiment_name)
-            hpo_service.instance.startExperiment(experiment_name)
+            response = hpo_service.instance.startExperiment(experiment_name)
+            if response:
+                return response
```
Comment (dinogun) [inline src/rest_service.py:267]: Can we explicitly return an error code here. It looks like we have several `if`'s which just assumes a null return

Comment (khansaad) [reply src/rest_service.py]: Removed redundant checks and added comment for clarity

File: src/hpo_service.py (hunk 3 above)
Comment (khansaad) [reply src/hpo_service.py:73]: Added

File: src/hpo_service.py (hunk 2 above)
Comment (khansaad) [reply src/hpo_service.py]: We are already validating when the request is received and return the response initially only. So this check isn't required again. Removed this and added the pre-check in gRPC as well

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**